Adding entity path and FullyQualifiedNamespace as readable properties in ServiceBusReceivedMessageContext

ki1729 opened the pull request:
# Description

Fixes https://github.com/Azure/azure-sdk-for-java/issues/29089.

# All SDK Contribution checklist:
- [x] **The pull request does not introduce [breaking changes]**
- [x] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [x] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md).**

## [General Guidelines and Best Practices](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md#developer-guide)
- [x] Title of the pull request is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

azure-sdk commented:
**API change check**

APIView has identified API level changes in this PR and created following API reviews.

[azure-messaging-servicebus](https://apiview.dev/Assemblies/Review/aa04238b10554e9ab8df210daf788450)

anuchandy reviewed (commented):
Looking at the .NET counterpart, it seems the way this is implemented is by exposing these two properties on the object emitted to the handler. Is there a reason we cannot align with .NET Api, (maybe I'm missing some context)

ki1729 commented:
> Looking at the .NET counterpart, it seems the way this is implemented is by exposing these two properties on the object emitted to the handler. Is there a reason we cannot align with .NET Api, (maybe I'm missing some context)

Ha, nope no missing context here. I saw that the object passed to the handler already has referrence to the receiverClient, from where entities can be read, so I added it to the processor client itself. The thing I skipped over was the ref in the object being private 😆 .  Fixed it

anuchandy reviewed (commented):
thank you
